### ISSUE TITLE
Cannot start the kafka-init with 0.5.5 release

### ISSUE THREAD

**Issue by nivemaham** [RADAR-base/RADAR-Schemas#200]
When I tried to deploy the latest version (0.5.5) of radar-schemas, I am facing the following issue.
This is running on the kubernetes instance.
It looks like it has successfully created all topics. However, registering schemas was not successful.
I think there are couple of issues.
One is, It cannot set the schema compatibility. This is also the case with 0.5.4. But the topic creation and schema registration is successful.
Is it acceptable to proceed without setting a schema combatibilty?

Second is the schema registration fails for 0.5.5 (on topic `questionnaire_ari_self`). Could this be specific to the topic/schema? 
Since it says timeout, It is hard for me to come to a conclusion if it is because of the schema or it cannot reach the schema-registry for other reasons.
This is a blocker for radar-ad deployments. So I request your support to fix this ASAP.

@blootsvoets @yatharthranjan @mpgxvii @afolarin 
```
Topics created.
Registering RADAR-base schemas...
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Setting compatibility to NONE
[main] ERROR org.radarcns.schema.registration.SchemaRegistry - Error changing compatibility level to NONE
java.net.SocketTimeoutException: timeout
	at okio.SocketAsyncTimeout.newTimeoutException(Okio.kt:159)
	at okio.AsyncTimeout.exit$jvm(AsyncTimeout.kt:203)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:163)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.kt:349)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:222)
	at okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.kt:210)
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:181)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:105)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:82)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:37)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:82)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
	at okhttp3.RealCall.execute(RealCall.kt:66)
	at org.radarbase.producer.rest.RestClient.request(RestClient.java:79)
	at org.radarcns.schema.registration.SchemaRegistry.putCompatibility(SchemaRegistry.java:141)
	at org.radarcns.schema.registration.SchemaRegistry$RegisterCommand.execute(SchemaRegistry.java:175)
	at org.radarcns.schema.CommandLineApp.main(CommandLineApp.java:167)
Caused by: java.net.SocketException: Socket closed
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:183)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
	at okio.InputStreamSource.read(Okio.kt:102)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:159)
	... 24 more
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_acceleration schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1Acceleration
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_battery_level schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1BatteryLevel
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_blood_volume_pulse schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1BloodPulseWave
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_energy schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1Energy
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_galvanic_skin_response schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1GalvanicSkinResponse
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_heartrate schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1HeartRate
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_heartrate_variability schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1HeartRateVariability
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_led_current schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1LedCurrent
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_oxygen_saturation schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1OxygenSaturation
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_ppg_raw schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1PpgRaw
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_respiration_rate schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1RespirationRate
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_biovotion_vsm1_temperature schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.biovotion.BiovotionVsm1Temperature
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_bittium_faros_acceleration schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.bittium.BittiumFarosAcceleration
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_bittium_faros_battery_level schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.bittium.BittiumFarosBatteryLevel
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_bittium_faros_ecg schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.bittium.BittiumFarosEcg
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_bittium_faros_inter_beat_interval schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.bittium.BittiumFarosInterBeatInterval
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_bittium_faros_temperature schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.bittium.BittiumFarosTemperature
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_acceleration schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4Acceleration
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_battery_level schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4BatteryLevel
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_blood_volume_pulse schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4BloodVolumePulse
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_electrodermal_activity schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4ElectroDermalActivity
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_inter_beat_interval schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4InterBeatInterval
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_sensor_status schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4SensorStatus
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_empatica_e4_temperature schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.empatica.EmpaticaE4Temperature
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_local_weather schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.weather.LocalWeather
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_pebble_2_acceleration schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.pebble.Pebble2Acceleration
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_pebble_2_battery_level schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.pebble.Pebble2BatteryLevel
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_pebble_2_heartrate schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.pebble.Pebble2HeartRate
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_pebble_2_heartrate_filtered schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.pebble.Pebble2HeartRateFiltered
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_acceleration schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneAcceleration
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_battery_level schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneBatteryLevel
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_bluetooth_devices schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneBluetoothDevices
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_call schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneCall
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_contacts schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneContactList
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_gyroscope schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneGyroscope
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_light schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneLight
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_magnetic_field schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneMagneticField
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_ppg schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.ppg.PhoneCameraPpg
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_relative_location schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneRelativeLocation
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_sms schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneSms
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_sms_unread schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneSmsUnread
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_step_count schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneStepCount
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_usage_event schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneUsageEvent
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_phone_user_interaction schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.phone.PhoneUserInteraction
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic android_processed_audio schemas: org.radarcns.kafka.ObservationKey - org.radarcns.passive.opensmile.OpenSmile2PhoneAudio
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_device_info schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationDeviceInfo
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_external_time schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationExternalTime
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_record_counts schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationRecordCounts
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_server_status schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationServerStatus
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_time_zone schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationTimeZone
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic application_uptime schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.application.ApplicationUptime
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic notification_thinc_it schemas: org.radarcns.kafka.ObservationKey - org.radarcns.active.questionnaire.Questionnaire
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic notification_thinc_it schemas: org.radarcns.kafka.ObservationKey - org.radarcns.active.notification.Notification
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic questionnaire_app_event schemas: org.radarcns.kafka.ObservationKey - org.radarcns.monitor.questionnaire.QuestionnaireApplicationInteractionEvent
[main] INFO org.radarcns.schema.registration.SchemaRegistry - Registering topic questionnaire_ari_self schemas: org.radarcns.kafka.ObservationKey - org.radarcns.active.questionnaire.Questionnaire
[main] ERROR org.radarcns.schema.registration.SchemaRegistry - Failed to register schemas for topic questionnaire_ari_self
java.net.SocketTimeoutException: timeout
	at okio.SocketAsyncTimeout.newTimeoutException(Okio.kt:159)
	at okio.AsyncTimeout.exit$jvm(AsyncTimeout.kt:203)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:163)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.kt:349)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.kt:222)
	at okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.kt:210)
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.kt:181)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.kt:105)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:82)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:37)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:82)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
	at okhttp3.RealCall.execute(RealCall.kt:66)
	at org.radarbase.producer.rest.RestClient.request(RestClient.java:79)
	at org.radarbase.producer.rest.RestClient.requestString(RestClient.java:114)
	at org.radarbase.producer.rest.SchemaRetriever.addSchemaMetadata(SchemaRetriever.java:157)
	at org.radarcns.schema.registration.SchemaRegistry.registerSchema(SchemaRegistry.java:98)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:442)
	at java.base/java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:175)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:503)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:533)
	at org.radarcns.schema.registration.SchemaRegistry.registerSchemas(SchemaRegistry.java:90)
	at org.radarcns.schema.registration.SchemaRegistry$RegisterCommand.execute(SchemaRegistry.java:182)
	at org.radarcns.schema.CommandLineApp.main(CommandLineApp.java:167)
Caused by: java.net.SocketException: Socket closed
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:183)
	at java.base/java.net.SocketInputStream.read(SocketInputStream.java:140)
	at okio.InputStreamSource.read(Okio.kt:102)
	at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:159)
	... 40 more
```

**Comment by blootsvoets** [comment]
Some thoughts:

- is this fixed if Kafka init runs again?
- Could this be an HTTP/2.0 issue on the schema registry, where multiple threads are using the connection and one closes it? We could try setting OkHttp compatibility to HTTP/1.1 only.
- is this a simple socket timeout on busy services? We could set socket timeouts longer in OkHttp
- is the schema registry setting up too slow under high memory pressure? The pod and JVM should get enough memory. Or is k8s moving the pod around during execution of Kafka-init?

**Comment by nivemaham** [comment]
Thanks for the valuable inputs @blootsvoets .
It looks like it is a k8 issue.

> Some thoughts:
> 
>     * is this fixed if Kafka init runs again?
No. It doesn't resolve it. k8 restarts automatically until the pod is healthy and no luck
> 
>     * Could this be an HTTP/2.0 issue on the schema registry, where multiple threads are using the connection and one closes it? We could try setting OkHttp compatibility to HTTP/1.1 only.

Possibly. Is there a config that I can look into to do this? Could you please let me know?
> 
>     * is this a simple socket timeout on busy services? We could set socket timeouts longer in OkHttp
So that will be the Okhttp on schema-registry. Can we configure that? If you could share some pointers, it would be helpful
> 
>     * is the schema registry setting up too slow under high memory pressure? The pod and JVM should get enough memory. Or is k8s moving the pod around during execution of Kafka-init?
Possibly. It looks like a k8 related issue, since @yatharthranjan has tried the 0.5.5 on the Docker environment and it updated successfully. I will try increasing the memory.

**Comment by blootsvoets** [comment]
The connection timeout can be configured here:
https://github.com/RADAR-base/RADAR-Schemas/blob/857fedeff859608656eaff0e8e0b74cddaa7be75/java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/registration/SchemaRegistry.java#L69
https://github.com/RADAR-base/RADAR-Schemas/blob/857fedeff859608656eaff0e8e0b74cddaa7be75/java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/registration/SchemaRegistry.java#L71

Restricting the client to HTTP/1.1 would require changing radar-commons. The httpClient in `SchemaRegistry` can be restricted using
```java
RestClient.Builder clientBuilder = RestClient.global();
clientBuilder.httpClientBuilder().protocols(Arrays.asList(Protocol.HTTP_1_1));
httpClient = clientBuilder
    // other configuration 
    //.timeout
    .build()
```

**Comment by blootsvoets** [comment]
Should be fixed in 0.7.0. Please reopen if this is not the case.
